editor/src/components/canvas/canvas-strategies/strategies/reparent-strategy-helpers.ts

```diff
@@ -499,69 +503,89 @@ function drawTargetRectanglesForChildrenOfElement(
-    pseudoElementAfter,
+    pseudoElementRightOrBottom,
   ]
 
-  let flexInsertionTargets: Array<CanvasRectangle> = []
+  let flexInsertionTargets: Array<{ rect: CanvasRectangle; insertionIndex: number }> = []
 
   if (targetRectangleSize === 'padded-edge') {
-    for (let index = 0; index < childrenBoundsAlongAxis.length - 1; index++) {
-      const start = childrenBoundsAlongAxis[index].end
-      const end = childrenBoundsAlongAxis[index + 1].start
-
-      const normalizedStart = Math.min(start, end)
-      const normalizedEnd = Math.max(start, end)
+    for (let index = 0; index < childrenBoundsAlongAxis.length; index++) {
+      const bounds = childrenBoundsAlongAxis[index]
 
-      const paddedStart = normalizedStart - ExtraPadding
-      const paddedEnd = normalizedEnd + ExtraPadding
+      const normalizedStart = Math.min(bounds.start, bounds.end)
+      const normalizedEnd = Math.max(bounds.start, bounds.end)
 
       flexInsertionTargets.push(
-        rectFromTwoPoints(
-          {
-            [leftOrTop]: paddedStart,
-            [leftOrTopComplement]: parentBounds[leftOrTopComplement],
-          } as any as CanvasPoint, // TODO improve my type
-          {
-            [leftOrTop]: paddedEnd,
-            [leftOrTopComplement]:
-              parentBounds[leftOrTopComplement] + parentBounds[widthOrHeightComplement],
-          } as any as CanvasPoint, // TODO improve my type
-        ),
+        {
+          insertionIndex: forwardsOrBackwards === 'forward' ? bounds.index : bounds.index + 1,
+          rect: rectFromTwoPoints(
+            {
+              [leftOrTop]: normalizedStart - ExtraPadding,
+              [leftOrTopComplement]: parentBounds[leftOrTopComplement],
+            } as any as CanvasPoint, // TODO improve my type
+            {
+              [leftOrTop]: Math.min(
+                normalizedStart + ExtraPadding,
+                normalizedStart + bounds.size / 2,
+              ),
+              [leftOrTopComplement]:
+                parentBounds[leftOrTopComplement] + parentBounds[widthOrHeightComplement],
+            } as any as CanvasPoint, // TODO improve my type
+          ),
+        },
+        {
+          insertionIndex: forwardsOrBackwards === 'forward' ? bounds.index + 1 : bounds.index,
+          rect: rectFromTwoPoints(
+            {
+              [leftOrTop]: Math.max(normalizedEnd - ExtraPadding, normalizedEnd - bounds.size / 2),
+              [leftOrTopComplement]: parentBounds[leftOrTopComplement],
+            } as any as CanvasPoint, // TODO improve my type
+            {
+              [leftOrTop]: normalizedEnd + ExtraPadding,
+              [leftOrTopComplement]:
+                parentBounds[leftOrTopComplement] + parentBounds[widthOrHeightComplement],
+            } as any as CanvasPoint, // TODO improve my type
+          ),
+        },
       )
     }
   } else {
+    // TODO BEFORE MERGE write a breaking test!!!!
```

Comment:
is this written? If yes, delete the comment

Reply:
fixed